src/plugins/condition/StyleRuleManager.js

```diff
@@ -27,24 +27,32 @@ export default class StyleRuleManager extends EventEmitter {
         super();
         this.openmct = openmct;
         this.callback = callback;
+        this.suppressSubscriptionOnEdit = suppressSubscriptionOnEdit;
         this.refreshData = this.refreshData.bind(this);
         this.toggleSubscription = this.toggleSubscription.bind(this);
-        if (suppressSubscriptionOnEdit) {
-            this.openmct.editor.on('isEditing', this.toggleSubscription);
-            this.isEditing = this.openmct.editor.editing;
-        }
+        this.addBoundsAndEditSubscriptions(styleConfiguration);
 
         if (styleConfiguration) {
             this.initialize(styleConfiguration);
             if (styleConfiguration.conditionSetIdentifier) {
-                this.openmct.time.on("bounds", this.refreshData);
                 this.subscribeToConditionSet();
             } else {
                 this.applyStaticStyle();
             }
         }
     }
 
+    addBoundsAndEditSubscriptions(styleConfiguration) {
```

Comment:
no arrow functions 😀
I've also checked the length of the listeners, and the number of StyleRuleManager listeners for the editor remains constant.